### PR TITLE
Add EU accounts support and trash_clear

### PR DESCRIPTION
**Pull request by qo4on**
Flags are not defined after import
```
from pcloud import PyCloud
print(O_CREAT)  # not defined
```

### CODE REVIEW

**Comment by tomgross** [comment]
You have to import the flags obviously:

```
from pcloud import O_CREAT
```

File: src/pcloud/api.py [hunk 1]
```diff
@@ -38,9 +38,18 @@ def main():
 
 class PyCloud(object):
 
-    endpoint = "https://api.pcloud.com/"
+    O_WRITE = int("0x0002", 16)
+    O_CREAT = int("0x0040", 16)
+    O_EXCL = int("0x0080", 16)
+    O_TRUNC = int("0x0200", 16)
+    O_APPEND = int("0x0400", 16)
```
Comment (tomgross) [inline src/pcloud/api.py:45]: Please remove the flags here. They are available on top level and need to be imported explicitly, if needed.

File: src/pcloud/api.py [hunk 2]
```diff
@@ -38,9 +38,18 @@ def main():
...
+
+    def __init__(self, username, password, eu_server=False):
+        if eu_server:
+            self.endpoint = "https://eapi.pcloud.com/"
+        else:
+            self.endpoint = "https://api.pcloud.com/"
```
Comment (tomgross) [inline src/pcloud/api.py:51]: The idea is not bad, but it makes the the tests fail and it does not scale. How about a endpoint dict with (api, eapi, unittest) as keys and the according URLs. This way it is easy to add more and the tests won't break.

File: src/pcloud/api.py [hunk 3]
```diff
@@ -158,6 +167,14 @@ def renamefile(self, **kwargs):
...
         return self._do_request("stat", **kwargs)
 
+    def trash_list(self, **kwargs):
+        return self._do_request("trash_list", **kwargs)
+
+    def trash_clear(self, **kwargs):
+        if not any(['fileid' in kwargs, 'folderid' in kwargs]):
+            kwargs['folderid'] = 0
+        return self._do_request("trash_clear", **kwargs)
+
```
Comment (tomgross) [inline src/pcloud/api.py:177]: These changes look good. Thanks!

File: src/pcloud/api.py (hunk 2 above)
Comment (qo4on) [reply src/pcloud/api.py]: This way `pycloud` will work with all endpoints and you don't have to make changes at all:
```
def __init__(self, username, password, endpoint="api"):
    self.endpoint = "https://" + endpoint + ".pcloud.com/"
```

**Comment by asanakoy** [comment]
Thanks for the fixes @qo4on !

**Comment by tomgross** [comment]
Implemented in https://github.com/tomgross/pycloud/commit/46e104020b9b01bf5d7036b2331c91042dcccfa7

**Pull request closed without merging**